sympy/diffgeom/tests/test_diffgeom.py

```diff
@@ -36,6 +36,34 @@ def test_R2():
         assert m == R2_p.coord_tuple_transform_to(
             R2_r, R2_r.coord_tuple_transform_to(R2_p, m)).applyfunc(simplify)
 
+
+def test_inverse_transformations():
+    p, q, r, s = symbols('p q r s')
+
+    relations_quarter_rotation = {
+        ('first', 'second'): (q, - p)
+    }
+
+    R2_pq = CoordSystem('first', R2_origin, [p, q], relations_quarter_rotation)
+    R2_rs = CoordSystem('second', R2_origin, [r, s], relations_quarter_rotation)
+
+    # The transform method should derive the inverse transformation if not given explicitly
+    assert R2_rs.transform(R2_pq) == Matrix([[- R2_rs.symbols[1]], [R2_rs.symbols[0]]])
+
+    a, b = symbols('a b', positive=True)
+    relations_uninvertible_transformation = {
+        ('first', 'second'): (- a,)
+    }
+
+    R2_a = CoordSystem('first', R2_origin, [a], relations_uninvertible_transformation)
+    R2_b = CoordSystem('second', R2_origin, [b], relations_uninvertible_transformation)
+
+    # The transform method should throw if it cannot invert the coordinate transformation.
+    # This transformation is uninvertible because there is no positive a, b satisfying a = -b
+    with raises(NotImplementedError):
+        R2_b.transform(R2_a)
```

Comment:
Please remove the space between the negative sign and the expression (`- a` -> `-a`)

Reply:
Done